src/clojure/clj_yaml/core.clj

```diff
@@ -157,3 +157,23 @@
                             :allow-recursive-keys allow-recursive-keys
                             :allow-duplicate-keys allow-duplicate-keys)
                  string) keywords))
+
+;; From https://github.com/metosin/muuntaja/pull/94/files
+(defn generate-stream
+  "Dump the content of data as yaml into writer."
+  [writer data & opts]
+  (.dump ^Yaml (apply make-yaml opts) (encode data) writer))
+
+(defn parse-stream
+  [^java.io.Reader reader & {:keys [keywords eof] :or {keywords true} :as opts}]
+  (letfn [(parse []
+            (decode (.load ^Yaml (apply make-yaml (flatten (seq opts)))
+                           reader) keywords))]
+    (if (nil? eof)
+      (parse)
+      (try
+        (parse)
+        (catch org.yaml.snakeyaml.error.YAMLException e
+          (if (instance? java.io.EOFException (.getCause e))
```

Comment:
This might not be correct: e.g. if you parse `{"a` you will probably get an `EOFException` but only returning the `eof` value would not be the correct thing to do I think.

Reply:
Compare:

``` 
user=> (read-string {:eof 1} "")
1
user=> (read-string {:eof 1} "(")
Execution error at user/eval148 (REPL:1).
EOF while reading
```

Reply:
> but only returning the eof value would not be the correct thing to do I think.

Ah, I thought the idea of `:eof` was that instead of throwing exceptions we'd return values as they are easier to work with instead (in which case, actually the current code is not quite correct. it may still throw `org.yaml.snakeyaml.parser.ParserException` or `org.yaml.snakeyaml.scanner.ScannerException`).

What behaviour would you suggest in general here?

Reply:
The `:eof` option is used in Clojure read functions to return the end of file as a value instead of as an exception, but this doesn't mean an invalid parse. If the underlying yaml lib doesn't support this, via some predicate, for example, I suggest simply not supporting it in the Clojure API.

Reply:
Cool, makes sense.

Looking at how SnakeYAML uses eof I don't think this is possible: https://bitbucket.org/asomov/snakeyaml/src/1ae187e03a4347c3495dda59b7b2c1fdcffc56b3/src/main/java/org/yaml/snakeyaml/reader/StreamReader.java#lines-48

I've removed it :slightly_smiling_face: